### DISCUSSION TITLE
Mesh in UI

### DISCUSSION THREAD

**Discussion by andrewbaxter** [bevyengine/bevy#14061] [Q&A]
I have some vector elements I want to include in my UI - boxes for text, scoreboards, etc. I created a `MaterialMesh2dBundle` in a `NodeBundle` but nothing's getting rendered.  I've googled around but haven't been able to find anything relevant. How can I achieve this?

**Comment by bugsweeper** [comment]
Can you add minimal reproducable example, which shows how exactly you are trying to mix meshes and Nodes?

**Reply by andrewbaxter** [reply]
I don't have a minimal example offhand, but are you saying this is expected to work? If so I'll try to come up with a minimal example.

**Reply by bugsweeper** [reply]
I'm not saying this is expected to work, I'm saying that for using Node's tree we use NodeBundle with defined style (which could be default, but there IS style for each node), UiSurface uses this style for rendering. I don't understand how could you set MaterailMesh2dBundle as Node with style set. Or you just spawn as simply bundle hoping for the best?

**Reply by bugsweeper** [reply]
I just wanted to ask what exactly you did.

**Reply by andrewbaxter** [reply]
Ah, I see. I did
```
commands.spawn_empty().insert(NodeBundle{...}).with_children(|commands| {
    commands.spawn_empty().insert(MaterialMesh2dBundle{
        mesh: Mesh2dHandle(meshes.add(Circle { radius: 50.0 })),
        material: materials.add(Color::BLUE),
        .. Default::default()
    });
});
```
roughly.
